apps/condo/domains/organization/utils/accessSchema.js

```diff
@@ -1,191 +1,297 @@
-    if (isEmpty(permissions) && employee) {
-        return true
-    }
+    return pick(roleRecord, permissionKeys)
+}
 
-    if (!employee || !employee.role) {
-        return false
+/**
+ * Clear cache data for all users employed in organization
+ * @param {string} organizationId - id of target organization
+ * @param {string | undefined} roleId - filter employees by roles
+ * @returns {Promise<void>}
+ */
+async function resetOrganizationEmployeesCache (organizationId, roleId = undefined) {
+    const employeesFilter = {
+        deletedAt: null,
+        organization: { id: organizationId },
+    }
+    if (roleId) {
+        employeesFilter.role = { id: roleId }
     }
 
-    const [employeeRole] = await find('OrganizationEmployeeRole', {
-        id: employee.role,
-        organization: { id: organizationId },
-    })
+    const employees = await find('OrganizationEmployee', employeesFilter)
 
-    if (!employeeRole) return false
-    return permissions.every((permission) => !!employeeRole[permission]) || false
+    if (employees.length) {
+        const cacheKeys = employees.map(employee => _getUserOrganizationsCacheKey({ id: employee.user }))
+        await _redisClient.del(cacheKeys)
+    }
 }
 
-async function checkRelatedOrganizationPermission (userId, organizationId, permission) {
-    return checkRelatedOrganizationPermissions(userId, organizationId, [permission])
+/**
+ * Clear cache data for specified user
+ * @param {string} userId
+ * @returns {Promise<void>}
+ */
+async function resetUserEmployeesCache (userId) {
+    const cacheKey = _getUserOrganizationsCacheKey({ id: userId })
+    await _redisClient.del(cacheKey)
 }
 
 /**
- * @param {string} userId
- * @param {string} organizationId
- * @param {string[]} permissions
- * @return {Promise<boolean>}
- */
-async function checkRelatedOrganizationPermissions (userId, organizationId, permissions) {
-    if (!userId || !organizationId) return false
-    const [organizationLink] = await find('OrganizationLink', {
-        from: queryOrganizationEmployeeFor(userId),
-        to: { id: organizationId },
+ * Information about single organization, in which user is employed, and its child organizations
+ * @typedef {Object} UserOgranizationInfo
+ * @property {Record<string, boolean>} permissions - user permissions in organization
+ * @property {Array<string>} childOrganizations - list of child organizations to which user share permissions via OrganizationLink
+ */
+
+/**
+ * Information about all organizations user has access to
+ * @typedef {Object} UserOgranizationsCache
+ * @property {number} dv - cache entry data version
+ * @property {Record<string, UserOgranizationInfo>} organizations - record where keys are organization ids and values are info it
+ */
+
+/**
+ * Obtains user organizations info via caching or sub-querying
+ * @param {{ id: string }} user - user object
+ * @returns {Promise<UserOgranizationsCache>}
+ * @private
+ */
+async function _getUserOrganizations (user) {
```

Comment:
I don't think that it's necessary to mark function as 'private' here. It's used only in concrete module and not exported

Reply:
Someone can add it to export. I’d you really don’t want it it’s ok to mark it as private